src/meta/client/src/grpc_client.rs

```diff
@@ -780,81 +779,31 @@ impl MetaGrpcClient {
     }
 
     #[tracing::instrument(level = "debug", skip(self, v))]
-    pub(crate) async fn do_write<T, R>(&self, v: T) -> Result<R, KVAppError>
-    where
-        T: RequestFor<Reply = R> + Into<MetaGrpcWriteReq>,
-        R: DeserializeOwned,
-    {
-        let act: MetaGrpcWriteReq = v.into();
-
-        debug!(req = debug(&act), "MetaGrpcClient::do_write request");
-
-        let req: Request<RaftRequest> = act.clone().try_into().map_err(|e| {
-            MetaNetworkError::InvalidArgument(InvalidArgument::new(e, "fail to encode request"))
-        })?;
-
-        debug!(
-            req = debug(&req),
-            "MetaGrpcClient::do_write serialized request"
-        );
-
-        let req = common_tracing::inject_span_to_tonic_request(req);
-
-        let mut client = self.make_client().await?;
-        let result = client.write_msg(req).await;
-        let result: Result<RaftReply, Status> = match result {
-            Ok(r) => Ok(r.into_inner()),
-            Err(s) => {
-                if status_is_retryable(&s) {
-                    self.mark_as_unhealthy().await;
-                    let mut client = self.make_client().await?;
-                    let req: Request<RaftRequest> = act.try_into().map_err(|e| {
-                        MetaNetworkError::InvalidArgument(InvalidArgument::new(
-                            e,
-                            "fail to encode request",
-                        ))
-                    })?;
-                    let req = common_tracing::inject_span_to_tonic_request(req);
-                    Ok(client.write_msg(req).await?.into_inner())
-                } else {
-                    Err(s)
-                }
-            }
-        };
-
-        let raft_reply = result?;
-
-        let res: Result<R, KVAppError> = raft_reply.into();
-
-        res
-    }
-
-    #[tracing::instrument(level = "debug", skip(self, v))]
-    pub(crate) async fn do_read<T, R>(&self, v: T) -> Result<R, KVAppError>
+    pub(crate) async fn kv_api<T, R>(&self, v: T) -> Result<R, KVAppError>
```

Comment:
`kv_api` is more like a trait or struct, not a function name. How about `send` or `request` (like in HTTP client)?

Reply:
Well... there are several RPC in this struct: 
- KV oriented: get, list, upsert, mget
- transactional API: `txn(condition1, condition2..., action1, action2...)`
- admin API, such as that returning a member list.
- ...

`send()` may not be specific enough to tell what it does, IMHO.